Makefile

```diff
@@ -138,17 +139,36 @@ properclean: ## Completely clean the repository's build artifacts.
 	rm -rf $(DISTDIR) $(TESTDIR)
 	$(DOCKER) rmi $(IMAGE)
 
-.PHONY: git2go
-git2go: $(VENDORDIR)/github.com/libgit2/git2go/v31/static-build/install/lib/pkgconfig/libgit2.pc
-
-$(VENDORDIR)/github.com/libgit2/git2go/v31/static-build/install/lib/pkgconfig/libgit2.pc: $(VENDORDIR)/github.com/libgit2/git2go/v31/vendor/libgit2
-	$(MAKE) -C $(VENDORDIR)/github.com/libgit2/git2go/v31 install-static
+GIT2GO_VERSION ?= v31
 
-$(VENDORDIR)/github.com/libgit2/git2go/v31/vendor/libgit2: $(VENDORDIR)/github.com/libgit2/git2go
-	$(GIT) -C $(VENDORDIR)/github.com/libgit2/git2go/v31 submodule update --init --recursive
+.PHONY: git2go
+git2go: $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION)/static-build/install/lib/pkgconfig/libgit2.pc
+	(cd $(VENDORDIR)/github.com/libgit2/git2go/$(GIT2GO_VERSION) && $(GO) install --tags "static" ./...)
```

Comment:
This works for me locally:

```suggestion
	$(GO) install -tags static github.com/libgit2/git2go/v31/...
```

You can pass the `-x` flag to see all the executed commands and verify. Sample output with `-x`:

```
# get https://proxy.golang.org/golang.org/x/crypto/@v/v0.6.0.info: 200 OK (0.014s)
WORK=/tmp/go-build3675054337
mkdir -p $WORK/b005/
cat >/tmp/go-build3675054337/b005/importcfg << 'EOF' # internal
# import config
EOF
cd /home/acotten/git/unikraft/kraftkit
/home/acotten/.local/share/go/pkg/tool/linux_amd64/compile -o $WORK/b005/_pkg_.a -trimpath "$WORK/b005=>" -p internal/goarch -std -+ -complete -buildid fcyczH0YXp2IMM4KyVvj/fcyczH0YXp2IMM4KyVvj -goversion go1.20.3 -c=4 -nolocalimports -importcfg $WORK/b005/importcfg -pack /home/acotten/.local/share/go/src/internal/goarch/goarch.go /home/acotten/.local/share/go/src/internal/goarch/goarch_amd64.go /home/acotten/.local/share/go/src/internal/goarch/zgoarch_amd64.go
mkdir -p $WORK/b012/


...
cd /home/acotten/git/unikraft/kraftkit/third_party/github.com/libgit2/git2go/v31
/home/acotten/.local/share/go/pkg/tool/linux_amd64/pack r $WORK/b001/_pkg_.a $WORK/b001/_x001.o $WORK/b001/_x002.o $WORK/b001/_x003.o $WORK/b001/_x004.o $WORK/b001/_x005.o $WORK/b001/_x006.o $WORK/b001/_x007.o $WORK/b001/_x008.o $WORK/b001/_x009.o $WORK/b001/_x010.o $WORK/b001/_x011.o $WORK/b001/_x012.o $WORK/b001/_x013.o $WORK/b001/_x014.o $WORK/b001/_x015.o $WORK/b001/_x016.o $WORK/b001/_x017.o $WORK/b001/_x018.o $WORK/b001/_x019.o $WORK/b001/_x020.o $WORK/b001/_x021.o $WORK/b001/_x022.o $WORK/b001/_x023.o $WORK/b001/_x024.o $WORK/b001/_x025.o $WORK/b001/_x026.o $WORK/b001/_x027.o $WORK/b001/_x028.o $WORK/b001/_x029.o $WORK/b001/_x030.o $WORK/b001/_x031.o $WORK/b001/_x032.o $WORK/b001/_x033.o $WORK/b001/_x034.o $WORK/b001/_x035.o $WORK/b001/_x036.o $WORK/b001/_x037.o $WORK/b001/_x038.o $WORK/b001/_x039.o $WORK/b001/_x040.o $WORK/b001/_x041.o $WORK/b001/_x042.o $WORK/b001/_x043.o $WORK/b001/_x044.o $WORK/b001/_x045.o $WORK/b001/_x046.o $WORK/b001/_x047.o $WORK/b001/_x048.o $WORK/b001/_x049.o # internal
/home/acotten/.local/share/go/pkg/tool/linux_amd64/buildid -w $WORK/b001/_pkg_.a # internal
cp $WORK/b001/_pkg_.a /home/acotten/.cache/go-build/11/11f08447963fcea23b664273ff35500e6a9b5474639dacf27ab3ccfc5f25bca6-d # internal
```
```

Reply:
Good catch! I tried a few variants but this one does the trick! Will incorporate in a fixup. :)